FreeRTOS/FreeRTOS-Plus-TCP
# IPv6/multi: recognise and treat IPv6 extension headers

Description
-----------
When we were doing the protocol testing, it was noticed that the IPv6 branch doesn't recognise and support so-called extension headers ( see [RFC 8200](https://datatracker.ietf.org/doc/html/rfc8200) ).

This PR adds a function `eHandleIPv6ExtensionHeaders()` that removes all extension headers, so that the packets can be processed normally.

Test Steps
-----------
I tested this PR by sending packets with hop-by-hop options to the IP-stack.

**To-do**: I could only test this PR while allowing checksum offloading. The function `usGenerateProtocolChecksum()` will still drop all packets containing hop-by-hop options.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
